embassy-nrf/src/nvmc.rs

```diff
@@ -55,6 +59,56 @@ impl<'d> Nvmc<'d> {
         let p = Self::regs();
         while p.ready.read().ready().is_busy() {}
     }
+
+    #[cfg(not(any(feature = "_nrf9160", feature = "_nrf5340")))]
+    fn wait_ready_write(&mut self) {
+        self.wait_ready();
+    }
+
+    #[cfg(any(feature = "_nrf9160", feature = "_nrf5340"))]
+    fn wait_ready_write(&mut self) {
+        let p = Self::regs();
+        while p.readynext.read().readynext().is_busy() {}
+    }
+
+    #[cfg(not(any(feature = "_nrf9160", feature = "_nrf5340")))]
+    fn erase_page(&mut self, page: u32) {
+        Self::regs().erasepage().write(|w| unsafe { w.bits(page) });
+    }
+
+    #[cfg(any(feature = "_nrf9160", feature = "_nrf5340"))]
+    fn erase_page(&mut self, page: u32) {
+        #[cfg(not(feature = "_nrf5340-net"))]
+        const FLASH_START_ADDR: u32 = 0;
+        #[cfg(feature = "_nrf5340-net")]
+        const FLASH_START_ADDR: u32 = 0x100_0000;
+
+        let first_page_word = (FLASH_START_ADDR + page * PAGE_SIZE as u32) as *mut u32;
```

Comment:
I think there is a bug here. **page** is the address off the page, and not the page number, thus the logic can be simplified to ```let firs_page_word = page as *mut u32```.

This also renders the **const FLASH_START_ADDR: ...** not needed.

Should the **page** argument be renamed to **page_addr** to improve clarity?

Reply:
Ah yes, you're right! Well spotted. `page_addr` is a better name yes.
Still didn't have the time to test this out. Hopefully before the next year.